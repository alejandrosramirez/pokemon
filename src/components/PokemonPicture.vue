<template>
	<div class="pokemon-picture-container">
		<img :src="pokemonImg" class="hidden-pokemon" alt="pokemon">

		<img v-if="showPokemon" :src="pokemonImg" class="fade-in" alt="pokemon">
	</div>
</template>

<script>
export default {
	props: {
		pokemonId: {
			type: Number,
			required: true,
		},
		showPokemon: {
			type: Boolean,
			required: true,
			default: false,
		},
	},
	name: "PokemonPicture",
	computed: {
		pokemonImg() {
			return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
		},
	},
};
</script>

<style scoped>
.pokemon-picture-container {
    height: 200px;
	display: flex;
	align-content: center;
	justify-content: center;
}

img {
	height: 200px;
	position: absolute;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.hidden-pokemon {
    filter: brightness(0);
}
</style>
