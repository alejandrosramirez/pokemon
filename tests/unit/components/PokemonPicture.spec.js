import { shallowMount } from "@vue/test-utils";

import PokemonPicture from "@/components/PokemonPicture.vue";

describe("PokemonPicture.vue", () => {
	test("Debe de hacer match con el snapshot", () => {
		const wrapper = shallowMount(PokemonPicture);
		expect(wrapper.html()).toMatchSnapshot();
	});

	test("Debe de mostrar la imagen oculta y el pokémon 100", () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: false,
			},
		});

		const [image1, image2] = wrapper.findAll("img");

		expect(image1.exists()).toBeTruthy();
		expect(image2).toBeUndefined();
		expect
	});

	test("Debe de mostrar la imagen del pokémon si showPokemon:true", () => {});
});
