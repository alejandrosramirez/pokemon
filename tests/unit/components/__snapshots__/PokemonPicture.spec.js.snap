// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture.vue Debe de hacer match con el snapshot 1`] = `
<div class="pokemon-picture-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/undefined.svg" class="hidden-pokemon" alt="pokemon">
  <!--v-if-->
</div>
`;
